services/web/server/src/simcore_service_webserver/wallets/_payments_handlers.py

```diff
@@ -143,3 +157,132 @@ async def cancel_payment(request: web.Request):
     )
 
     return web.HTTPNoContent(content_type=MIMETYPE_APPLICATION_JSON)
+
+
+#
+# Payment methods
+#
+
+
+class PaymentMethodsPathParams(WalletsPathParams):
+    payment_method_id: PaymentID
+
+
+@routes.post(
+    f"/{VTAG}/wallets/{{wallet_id}}/payments-methods:init",
+    name="init_creation_of_payment_method",
+)
+@login_required
+@permission_required("wallets.*")
+@handle_wallets_exceptions
+@requires_dev_feature_enabled
+async def init_creation_of_payment_method(request: web.Request):
+    """Triggers the creation of a new payment method.
+    Note that creating a payment-method follows the init-prompt-ack flow
+    """
+    req_ctx = WalletsRequestContext.parse_obj(request)
+    path_params = parse_request_path_parameters_as(WalletsPathParams, request)
+
+    with log_context(
+        _logger,
+        logging.INFO,
+        "Initated the creation of a payment-method for wallet %s",
+        f"{path_params.wallet_id=}",
+        log_duration=True,
+        extra=get_log_record_extra(user_id=req_ctx.user_id),
+    ):
+        initiated: PaymentMethodInit = await init_creation_of_wallet_payment_method(
+            request.app, user_id=req_ctx.user_id, wallet_id=path_params.wallet_id
+        )
+
+        return envelope_json_response(initiated, web.HTTPCreated)
+
+
+@routes.post(
+    f"/{VTAG}/wallets/{{wallet_id}}/payments-methods/{{payment_method_id}}:cancel",
+    name="cancel_creation_of_payment_method",
+)
+@login_required
+@permission_required("wallets.*")
+@handle_wallets_exceptions
+@requires_dev_feature_enabled
+async def cancel_creation_of_payment_method(request: web.Request):
```

Comment:
Q: this cancel why is reposnsible for calling it?
what if this is not called? We end up with payments that are half added, half initialised etc... ?
How do you cleanup in all the above situations?

Reply:
followed up on an discussion offline